lib/buildpacks_registry_api/cache_client.ex

```diff
@@ -0,0 +1,349 @@
+defmodule BuildpacksRegistryApi.CacheClient do
+  @moduledoc """
+  GenServer cached client implementation for the [Buildpacks Registry API](https://github.com/buildpacks/registry-api)
+
+  Default timeout is one (1) hour.
+  """
+
+  @name :buildpacks_registry_api_cache_client
+
+  use GenServer
+
+  require Logger
+
+  alias BuildpacksRegistryApi.Client
+
+  defmodule State do
+    @moduledoc false
+    defstruct cache_timeout: :timer.hours(1), cache: %{}
+  end
+
+  # Client functions
+  @spec start_link(any) :: :ignore | {:error, any} | {:ok, pid}
+  def start_link(_arg) do
+    Logger.info("Starting BuildpacksRegistryApi.CacheClient...")
+    GenServer.start_link(__MODULE__, %State{}, name: @name)
+  end
+
+  @doc """
+  Clears the cache.
+  """
+  @spec clear() :: :ok
+  def clear() do
+    GenServer.cast(@name, :clear)
+  end
+
+  @doc """
+  Gets the config for the cache client.
+  """
+  @spec config() :: map()
+  def config() do
+    GenServer.call(@name, :config)
+  end
+
+  @doc """
+  Sets the cache timeout in milliseconds.
+  """
+  @spec set_cache_timeout(integer) :: :ok
+  def set_cache_timeout(timeout_in_ms) do
+    GenServer.cast(@name, {:set_cache_timeout, timeout_in_ms})
+  end
+
+  @doc false
+  def get(path, query_params) do
+    GenServer.call(@name, {:return_cache_or_get, path, query_params})
+  end
+
+  @spec search(binary()) :: list(map())
+  @doc """
+
+  Searches the registry for a buildpacks that match a string term.
+
+  ## Examples
+
+    In the example below, we seach for ruby buildpacks and receive back a List of buildpack object maps:
+      BuildpacksRegistryApi.Client.search("ruby")
```

Comment:
Have you checked how these render in the docs?  My memory says either four spaces or fences required for syntax highlighting of code.

Reply:
Yep I think I messed these up when I adding some text. Fixed.